### PR TITLE
pep8, add doc and allow saving fig or not in script compare irfs

### PR DESCRIPTION
**Pull request by garciagenrique**

### CODE REVIEW

File: lstmcpipe/scripts/script_compare_irfs.py [hunk 1]
```diff
@@ -38,24 +55,33 @@ def main():
...
                         action='store_true',
                         dest='cta_north',
                         help='add CTA north performances curves'
                         )
 
+    parser.add_argument('--save-figure', '-s',
+                        action='store',
+                        type=lambda x: bool(strtobool(x)),
+                        dest='save_figure',
+                        help='Save resulting figure',
```
Comment (vuillaut) [inline lstmcpipe/scripts/script_compare_irfs.py:78]: implement with `action='store_false'` (remove type)

**Review by vuillaut — CHANGES_REQUESTED** [review]

**Comment by vuillaut** [comment]
> Why would you not want to save the figure? These scripts are not meant for interactive use, right?
> 
> Also the way I usually see this handled is to leave the `output_file` an empty string or `False` and then do:
> 
> ```python
> if output_file:
>     fig.savefig(output_file)
> else:
>     fig.show()
> ```
> 
> Does this even show the figure in the other case? Whats the point of calling the script?



> Why would you not want to save the figure? These scripts are not meant for interactive use, right?
> 
> Also the way I usually see this handled is to leave the `output_file` an empty string or `False` and then do:
> 
> ```python
> if output_file:
>     fig.savefig(output_file)
> else:
>     fig.show()
> ```
> 
> Does this even show the figure in the other case? Whats the point of calling the script?

I agree with `output_file=None` or `False` and the proposed implementation that would be much cleaner.

Then the script args can follow the same logic (save figure only if an outfile is passed).

Concerning the interactivity, I do use the **function** (not the script obviously) in notebooks at least  🤷‍♂️

**Comment by LukasNickel** [comment]
> Concerning the interactivity, I do use the **function** (not the script obviously) in notebooks at least  man_shrugging

Of course, interactive plots are nice and all.
But this is meant to be used on the cluster in a scheduled job. There is no reason for interactivity there.
(Thats my first impression, I might miss something, which is why I asked for the reason)

**Comment by vuillaut** [comment]
default None for outfile in function plot_comparison

**Comment by garciagenrique** [comment]
> Concerning the interactivity, I do use the function (not the script obviously) in notebooks at least man_shrugging



> > Concerning the interactivity, I do use the **function** (not the script obviously) in notebooks at least  man_shrugging
> 
> Of course, interactive plots are nice and all. But this is meant to be used on the cluster in a scheduled job. There is no reason for interactivity there. (Thats my first impression, I might miss something, which is why I asked for the reason)

Everything is indeed expected to be run at the cluster as batched job. But some examples of the school were calling the function interactively, for example :). And I have the impression that if we would like to start comparing plots, this functionality might be useful

**Comment by garciagenrique** [comment]
Thanks for the changes @vuillaut !

**Pull request merged**